HLTrigger/Tools/bin/hltTimingSummary.cpp

```diff
@@ -1880,16 +1880,16 @@ int main(int argc, char ** argv) {
       }
     }
     
-    char value[10] ; 
+    char value[330] ; 
```

Comment:
This is an error in GCC 7. It's a new flag. It's triggered for both, overflows and truncations. And yes, `snprintf` will just print just keep `n-1` chars from the format string (1 char is needed for NULL termination). The buffer needs to be big, because we are printing `double` here. I checked at least `DBL_MAX` and that seems to be 308+ chars.

Looks like the warning/errors is not triggered if one checks for return value of `snprintf`. In that case GCC assumes that developer detected truncation and handled it correctly.